examples/run_glue_trainer.py

```diff
@@ -0,0 +1,482 @@
+                copy = True
+                continue
+            elif line.strip() == 'finetune_hparams:':
+                copy = False
+                continue
+            elif copy:
+                outfile.write(line)
+    
+    hp = TrainerHparams.create(cli_args=False, f=tmp_file)
+    dataloader = hp.train_dataset.initialize_object(dataloader_hparams=hp.dataloader, batch_size=hp.train_batch_size)
+    dataloader_len = dataloader.num_samples // hp.train_batch_size
+    run_name = hp.run_name
+    save_folder = os.path.join(run_name, hp.save_folder) 
+
+    if training_scheme == 'all':  # extract run_name from trainer args for finetuning
+        # list and save checkpoint paths 
+        finetune_hparams.save_folder = save_folder
+        finetune_hparams.finetune_ckpts = get_ckpt_names(hp, run_name, dataloader_len)
+        
+    # call via composer to ensure pretraining done distributedly across all available GPUs
+    subprocess.run(args=['composer', training_script, '-f', tmp_file, '--save_folder', save_folder], check=True)
+
+def run_finetuner(training_scheme: str, file: str, save_locally: bool, save_folder: str, finetune_hparams, glue_metrics: GlueState) -> None:
+    """Logic for handling a finetuning job spawn based on storage and training settings."""
+    # set automatic load and save paths
+    all_ckpts_list = finetune_hparams.finetune_ckpts
+
+    # finetune on every pretrained checkpoint
+    for ckpt_filename in all_ckpts_list:
+        parent_ckpt = ckpt_filename  # necessary for logging
+
+        task_to_save_ckpt = {'cola': False, 'sst-2': False, 'qqp': False, 'qnli': False, 'mnli': True} 
+        spawn_finetuning_jobs(task_to_save_ckpt,
+                              ckpt_filename,
+                              save_folder,
+                              glue_metrics,
+                              file,
+                              save_locally=save_locally,
+                              parent_ckpt=parent_ckpt,
+                              load_ignore_keys=['state/model/model.classifier*'])
+
+        # finetune on inference tasks using last mnli checkpoint
+        ckpt_filename = f'{save_folder}/mnli/latest-rank0'   
+
+        mnli_task_to_save_ckpt = {'rte': False, 'mrpc': False, 'stsb': False}
+        # delete finetuning head to reinitialize number of classes
+        spawn_finetuning_jobs(
+            mnli_task_to_save_ckpt,
+            ckpt_filename,
+            save_folder,
+            glue_metrics,
+            file,
+            save_locally=save_locally,
+            parent_ckpt=parent_ckpt,
+            load_ignore_keys=['state/model/model.classifier*'],
+        )
+
+
+def _main() -> None:
+    warnings.formatwarning = warning_on_one_line
+
+    if len(sys.argv) == 1:
+        sys.argv = [sys.argv[0], '--help']
+
+    file, training_scheme = get_args()
+    finetune_hparams, save_locally = get_finetune_hparams(training_scheme)
+
+    # Pretrain
+    if training_scheme != 'finetune':
```

Comment:
nit: for readability, can we do 
```suggestion
    if training_scheme in ('pretrain', 'all'):
```

And same thing below on line 472.